tests/functional/plugincheck.js

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global casper */
+'use strict';
+
+var config = require('../lib/config');
+var helpers = require('../lib/helpers');
+var path = '/plugincheck/';
+var url = config.base() + path;
+
+casper.options.pageSettings.loadPlugins = true;
+
+casper.test.begin('PluginCheck, conditional content: ' + url, 6, function suite(test) {
+
+    casper.start();
+
+    casper.userAgent('Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:38.0) Gecko/20100101 Firefox/38.0');
+
+    casper.thenOpen(url, function() {
+
+        test.assertHttpStatus(200);
+
+        test.assertVisible('.plugin-status-container h2', 'Plugin Status heading exists');
+        test.assertNotVisible('.not-supported', 'Unsupported browser message not visible');
+
+        casper.userAgent('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36');
+    });
+
+    casper.thenOpen(url, function() {
+
+        test.assertHttpStatus(200);
```

Comment:
I was wondering about that. I was thinking that each block will reload the page so, perhaps just ensuring it actually loaded before stepping into the real tests might be a good 'sanity' check but, no reason to overdo things.

You reckon I should remove?